spec/support/shared_examples/persona.rb

```diff
@@ -0,0 +1,40 @@
+shared_examples :persona do
+  let(:approved_output_path) { 'spec/personas/approved_output' }
+  let(:intake) { create persona_name }
+  let(:efile_submission) { create :efile_submission, :accepted, :for_state, data_source: intake }
+
+  it 'generates identical filing PDF to approved output' do
+    generated_pdf = efile_submission.generate_filing_pdf
+    generated_pdf_hash = PdfForms.new.get_fields(generated_pdf).to_h { |field| [field.name, field.value] }
+    approved_pdf_path = "#{approved_output_path}/#{tax_year}/#{state_code}/#{persona_name}_return.pdf"
+    approved_pdf_hash = PdfForms.new.get_fields(File.open(approved_pdf_path)).to_h { |field| [field.name, field.value] }
+    expect(generated_pdf_hash).to match(approved_pdf_hash)
+  end
+
+  it 'generates identical submission bundle to approved output' do
+    create(:state_file_efile_device_info, :filled, :initial_creation, intake: intake, updated_at: Time.now - 1.minute)
+    create(:state_file_efile_device_info, :filled, :submission, intake: intake)
+    efile_submission.update(irs_submission_id: submission_id)
+    response = SubmissionBundle.new(efile_submission).build
+    expect(response.errors).to be_empty
+    expect(response.valid?).to be true
```

Comment:
look into whether you can give a custom failure message without creating a custom matcher